File: code_examples/Multi-Platform/src/f90/amb_ext1.f90
#include "param2.f90.h"

MODULE amb_ext1
  IMPLICIT NONE
  PRIVATE

  PUBLIC :: bmc

CONTAINS
SUBROUTINE bmc(nx, ll, b, c)
  USE amb_ext2, ONLY : mc

  IMPLICIT NONE
  INTEGER, INTENT(IN) :: nx
  REAL(8), INTENT(IN) :: ll
  REAL(8), DIMENSION(nx), INTENT(IN) :: b
  REAL(8), DIMENSION(nx), INTENT(INOUT) :: c

  INTEGER :: i
                
  DO i=1,nx
    c(i) = ll*b(i) + mc(MM, c(i))
  END DO
END SUBROUTINE bmc

END MODULE amb_ext1

File: code_examples/Multi-Platform/src/c/amb_ext1.c
#include "param2.h"
#include "amb_ext2.h"


void bmc(int nx, double ll, double *b, double *c) {
	// size and intent of array arguments for f2py
	// b :: nx, in
	// c :: nx, inout
	
	int i;

	for (i=0; i<nx; i++) {
		c[i] = ll*b[i] + mc(MM, c[i]);
	}
}
